### PR TITLE
add root to terminology

### LINKED ISSUE #5203: Add "root" to terminology
"root" shows up a lot in validator logs.  This is a Solana-specific term that deserves to be defined in https://solana-labs.github.io/book/terminology.html

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 see #5203

 #### Summary of Changes
 add root to terminology

Fixes #5203

### CODE REVIEW

File: book/src/terminology.md [hunk 1]
```diff
@@ -213,6 +213,10 @@ The public key of a [keypair](#keypair).
 Storage mining client, stores some part of the ledger enumerated in blocks and
 submits storage proofs to the chain. Not a full-node.
 
+#### root
```
Comment (garious) [inline book/src/terminology.md]: How about "root block"?

File: book/src/terminology.md [hunk 2]
```diff
@@ -213,6 +213,10 @@ The public key of a [keypair](#keypair).
...
+
+A [block](#block) or [slot](#slot) that has reached maximum [lockout](#lockout) on a validator.  The root is the highest block that is an ancestor of all active forks on a validator.  All ancestor blocks of a root are also transitively a root.  Any block that is not an ancestor or a descendant of the root is considered dead.  Once 2/3rds stake of validators on the network has set a particular block to root, the block is considered finalized.
```
Comment (garious) [inline book/src/terminology.md]: This sentence could a separate definition, a "dead block". Also, "dead" seems like an odd term for what you've described. "Orphaned"?

Comment (garious) [reply book/src/terminology.md]: The 2/3 sentence should be in the description of finality, not here

File: book/src/terminology.md (hunk 1 above)
Comment (rob-solana) [reply book/src/terminology.md]: I'm resistant.

We don't log or say `root block` anywhere, and there are no conflicts with any other `root Xs` yet.

File: book/src/terminology.md (hunk 2 above)
Comment (rob-solana) [reply book/src/terminology.md]: > This sentence could a separate definition, a "dead block". Also, "dead" seems like an odd term for what you've described. "Orphaned"?

I'd prefer something like "unreachable" or "prune-able", which shouldn't require another glossary entry, as we're not re-defining anything for our domain.

File: book/src/terminology.md (hunk 1 above)
Comment (garious) [reply book/src/terminology.md]: How about "root: see *rooted block*"

File: book/src/terminology.md (hunk 2 above)
Comment (rob-solana) [reply book/src/terminology.md]: > The 2/3 sentence should be in the description of finality, not here

Sounds good, I'll refactor around that.

File: book/src/terminology.md (hunk 1 above)
Comment (garious) [reply book/src/terminology.md]: Or "the highest *rooted block*

Comment (rob-solana) [reply book/src/terminology.md]: hm.  We say "a root", which is a useful thing in the context of a running validator.

still resistant ;)

**Pull request merged**